src/dot-staker.ts

```diff
@@ -175,10 +175,15 @@ export class DOTStaker {
    * Add proxy account to your stash account so it will have the permissions to nominate validators
    * @param vaultAccountId - stash vault account id
    * @param proxyAddress - DOT proxy address
+   * @param type - proxy type, defaults to "NonTransfer"
    */
-  public async addProxy(vaultAccountId: string, proxyAddress: string) {
+  public async addProxy(
+    vaultAccountId: string,
+    proxyAddress: string,
+    type: string = "NonTransfer"
```

Comment:
I would set an enum of possible values. Let's try to protect users from typos or incorrect values.